optimum/onnxruntime/modeling_ort.py

```diff
@@ -0,0 +1,681 @@
+    def from_transformers(
+        cls,
+        model_id: str,
+        save_dir: Union[str, Path] = default_cache_path,
+        use_auth_token: Optional[Union[bool, str, None]] = None,
+        revision: Optional[Union[str, None]] = None,
+        force_download: bool = True,
+        cache_dir: Optional[str] = None,
+        **kwargs,
+    ):
+        """
+        Converts a vanilla Transformers model into an optimized model using `transformers.onnx.export_onnx`.
+        Arguments:
+            model_id (:obj:`str` or :obj:`Path`):
+                Directory from which to load
+            save_dir (:obj:`str` or :obj:`Path`):
+                Directory where the onnx model should be saved, default to `transformers.file_utils.default_cache_path`, which is the cache dir for
+                transformers.
+            use_auth_token (:obj:`str` or :obj:`bool`):
+                Is needed to load models from a private repository
+            revision (:obj:`str`):
+                Revision is the specific model version to use. It can be a branch name, a tag name, or a commit id
+            cache_dir (:obj:`Union[str, Path]`, `optional`):
+                Path to a directory in which a downloaded pretrained model configuration should be cached if the
+                standard cache should not be used.
+            force_download (:obj:`bool`, `optional`, defaults to :obj:`False`):
+                Whether or not to force the (re-)download of the model weights and configuration files, overriding the
+                cached versions if they exist.
+            kwargs (:obj:`Dict`, `optional`)::
+                kwargs will be passed to the model during initialization
+        """
+
+        # create local save dir in cache dir
+        save_dir = Path(save_dir).joinpath(model_id)
+        save_dir.mkdir(parents=True, exist_ok=True)
+        kwargs["model_save_dir"] = save_dir
+
+        # reads pipeline task from OnnxForXXX class if available else tries to extract from hub
+        if cls.pipeline_task is not None:
+            task = cls.pipeline_task
+        else:
+            task = HfApi().model_info(model_id, revision=revision).pipeline_tag
+            if task in ["sentiment-analysis", "text-classification", "zero-shot-classification"]:
+                task = "sequence-classification"
+            elif task in ["feature-extraction", "fill-mask"]:
+                task = "default"
+        # 2. convert to temp dir
+        # FIXME: transformers.onnx conversion doesn't support private models
+        tokenizer = AutoTokenizer.from_pretrained(model_id)
+        model = FeaturesManager.get_model_from_feature(task, model_id)
+        _, model_onnx_config = FeaturesManager.check_supported_model_or_raise(model, feature=task)
+        onnx_config = model_onnx_config(model.config)
+
+        # export model
+        export(
+            preprocessor=tokenizer,
+            model=model,
+            config=onnx_config,
+            opset=13,
```

Comment:
Each `OnnxConfig` has a `default_onnx_opset` property, specifying an opset that should work for the architecture in question.
```suggestion
            opset=onnx_config.default_onnx_opset,
```